lib/ui/text/paragraph_builder.cc

```diff
@@ -187,19 +192,55 @@ ftl::RefPtr<ParagraphBuilder> ParagraphBuilder::create(
     double fontSize,
     double lineHeight,
     const std::string& ellipsis) {
-  return ftl::MakeRefCounted<ParagraphBuilder>(
-      encoded, fontFamily, fontSize, lineHeight, ellipsis);
+  return ftl::MakeRefCounted<ParagraphBuilder>(encoded, fontFamily, fontSize,
+                                               lineHeight, ellipsis);
 }
 
 ParagraphBuilder::ParagraphBuilder(tonic::Int32List& encoded,
                                    const std::string& fontFamily,
                                    double fontSize,
                                    double lineHeight,
                                    const std::string& ellipsis) {
+  if (!Paragraph::m_usingBlink) {
+    int32_t mask = encoded[0];
+    txt::ParagraphStyle style;
+    if (mask & psTextAlignMask)
+      style.text_align = txt::TextAlign(encoded[psTextAlignIndex]);
+
+    if (mask & (psFontWeightMask | psFontStyleMask | psFontFamilyMask |
+                psFontSizeMask)) {
+      if (mask & psFontWeightMask)
+        style.font_weight =
+            static_cast<txt::FontWeight>(encoded[psFontWeightIndex]);
+
+      if (mask & psFontStyleMask)
+        style.font_style =
+            static_cast<txt::FontStyle>(encoded[psFontStyleIndex]);
+
+      if (mask & psFontFamilyMask)
+        style.font_family = fontFamily;
+
+      if (mask & psFontSizeMask)
+        style.font_size = fontSize;
+    }
+
+    if (mask & psLineHeightMask)
+      style.line_height = lineHeight;
+
+    if (mask & psMaxLinesMask)
+      style.max_lines = encoded[psMaxLinesIndex];
+
+    if (mask & psEllipsisMask)
+      style.ellipsis = ellipsis;
+
+    m_paragraphBuilder.SetParagraphStyle(style);
+  }
```

Comment:
I'd put this in an "else".  We shouldn't do the blink-related work in the non-blink case.